boefjes/tools/upgrade_v1_16_0.py

```diff
@@ -0,0 +1,181 @@
+
+    bytes_client = get_bytes_client()
+    bytes_client.login()
+
+    total_failed = 0
+    total_processed = 0
+
+    organisations = organisation_repository.get_all()
+    logger.info("Processing %s organisations in total", len(organisations))
+
+    for organisation_id in organisations:
+        connector = get_octopoes_api_connector(organisation_id)
+        logger.info("Processing organisation [organization_id=%s]", organisation_id)
+
+        failed, processed = migrate_organisation(bytes_client, connector, organisation_id, valid_time)
+        total_failed += failed
+        total_processed += processed
+
+        logger.info("Processed organisation [total_processed=%s, total_failed=%s]", processed, failed)
+
+    logger.info("Finished migration [total_processed=%s, total_failed=%s]", total_processed, total_failed)
+
+    return total_processed, total_failed
+
+
+def migrate_organisation(bytes_client, connector, organisation_id, valid_time) -> tuple[int, int]:
+    """
+    For each organisation, we paginate through the origin API, find the matching normalizer meta in Bytes,
+    and set the source_method to the boefje id. Then update the origin, i.e. save it and delete the old one.
+    """
+
+    failed = 0
+    processed = 0
+
+    offset = 0
+    page_size = 200
+
+    boefjes_for_normalizer = collect_boefjes_per_normalizer()
+    bulk_updated_origins = []
+
+    while True:
+        # We loop through the paginated API until we reach the end
+
+        origins = connector.list_origins(
+            valid_time, method=[x for x in boefjes_for_normalizer], offset=offset, limit=page_size
+        )
+        logger.info("Processing %s origins", len(origins))
+
+        for origin in origins:
+            if origin.source_method is not None or origin.origin_type == OriginType.INFERENCE:
+                continue
+
+            if origin.method in boefjes_for_normalizer and len(boefjes_for_normalizer[origin.method]) == 1:
+                origin.source_method = boefjes_for_normalizer[origin.method][0].id
+                bulk_updated_origins.append(origin)
+                continue
+
+            try:
+                normalizer_meta = bytes_client.get_normalizer_meta(origin.task_id)
+                origin.source_method = normalizer_meta.raw_data.boefje_meta.boefje.id
+                bulk_updated_origins.append(origin)
+            except HTTPStatusError as error:
+                # We expect to find Declaration/Affirmations without a normalizer meta
+                if error.response.status_code == 404 and origin.method != "manual":
+                    logger.warning(
+                        "Could not find normalizer_meta [task_id=%s, method=%s, origin_type=%s]",
+                        origin.task_id,
+                        origin.method,
+                        origin.origin_type,
+                    )
+                elif error.response.status_code == 404:
+                    logger.info(
```

Comment:
If this is expected, shouldn't we log it at level debug?

```suggestion
                    logger.debug(
```

Reply:
I think in this case we rather show as much as possible for just this one script, instead of regretting not setting the level to debug before running it? For instance to verify if the right origins were skipped? I guess these should still show up on a second run though